server/src/main/scala/org/scassandra/server/priming/cors/CorsSupport.scala

```diff
@@ -0,0 +1,43 @@
+package org.scassandra.server.priming.cors
+
+import com.typesafe.scalalogging.LazyLogging
+import org.scassandra.server.ScassandraConfig
+import spray.http.HttpHeaders._
+import spray.http.HttpMethods._
+import spray.http.{AllOrigins, HttpMethods, HttpResponse}
+import spray.routing._
+
+/**
+  * A mixin to provide support for providing CORS headers as appropriate
+  */
+trait CorsSupport {
+  this: HttpService with LazyLogging =>
+
+  private val allowOriginHeader = `Access-Control-Allow-Origin`(AllOrigins)
+  private val optionsCorsHeaders = List(
+    `Access-Control-Allow-Headers`(
+      "Origin, X-Requested-With, Content-Type, Accept, Accept-Encoding, Accept-Language, Host, Referer, User-Agent"
+    ),
+    `Access-Control-Max-Age`(60 * 60 * 24 * 20) // cache pre-flight response for 20 days
+  )
+
+  def cors[T]: Directive0 = mapRequestContext {
+    context => context.withRouteResponseHandling {
+      // If an OPTIONS request was rejected as 405, complete the request by responding with the
+      // defined CORS details and the allowed options grabbed from the rejection
+      case Rejected(reasons) if
```

Comment:
I'm not entirely sure how this works since I haven't used spray that much, is this intercepting what would be a 405 response and injecting the appropriate response with all the headers?